cmd/controller/main.go

```diff
@@ -87,48 +87,42 @@ func run(cmd *cobra.Command, args []string) {
 	if err != nil {
 		log.Fatalf("failed to create the cluster manager: %v", err)
 	}
-	ctlrNamespace, err := initializeControllerNamespace()
+	ctlrNamespace, err := initializeControllerNamespace(mgr.GetClient())
 	if err != nil {
 		log.Fatalf("failed to get controller namespace: %s", err)
 	}
-	log.Info("initializing machine actuator")
-	machineActuator, err := wks.NewMachineActuator(wks.MachineActuatorParams{
-		EventRecorder:       mgr.GetRecorder(wks.ProviderName + "-controller"),
+
+	log.Info("registering scheme for all resources")
+	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Fatal(err)
+	}
+	if err := clusterv1.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Fatal(err)
+	}
+
+	log.Info("registering controllers to the cluster manager")
+	clusterReconciler, err := wks.NewClusterReconciler(mgr.GetClient(), mgr.GetEventRecorderFor(wks.ProviderName+"-controller"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err = clusterReconciler.SetupWithManager(mgr, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
```

Comment:
Can the reconcile occur in parallel or does it have to be a singleton? A lot of the other provider expose a `concurrency` flag.